app/components/PartialChallengeDetailsView.js

```diff
@@ -41,46 +43,19 @@ export default class PartialChallengeDetailsView extends Component {
 
   render() {
     return (
-      <View style={styles.container}>
-        {!this.props.challenge ? 
-          <ActivityIndicator
-            animating={true}
-            style={[styles.centering, {height: 80}]}
-            size="large"
-          /> : 
-          <MapView
-            ref = { (MapRef) => {if( MapRef !=null ) { MapRef.fitToElements(true) }} }
-            style={styles.map}
-            cacheEnabled={true}
-            loadingEnabled
-           >
-            {this.props.challenge.locations.map(location => (
-              <MapView.Marker 
-                key={location.title}
-                identifier={location.challengeId}
-                coordinate={{latitude: location.latitude, longitude: location.longitude}}
-                title={location.title}
-                image={secondaryMarkerImage}
-              />
-            ))}
-           </MapView>
-        }
-        <Text>
-          <Text style={{fontWeight: 'bold'}}>Challenge: </Text>
-          {this.props.challenge.title}
-        </Text>
-        <Text>
-          <Text style={{fontWeight: 'bold'}}>GPS Points: </Text>
-          <Text>{this.props.challenge.locations.length}</Text>
-        </Text>
-        <Text>
-          <Text style={{fontWeight: 'bold'}}>Completed by: </Text>
-          {this.props.challenge.completedBy.length} user{this.props.challenge.completedBy.length > 1 && <Text>s</Text>}
-        </Text>
-        <Button 
-          onPress={this.goToChallengeDetailScene}
-          title="Go to Challenge Detail Scene"
-          />
+      <View style={styles.challengeDetails}>
+        <View>
+          <Text style={[styles.caption,styles.title]}>{this.props.challenge.title}</Text>
+          <TouchableHighlight
+            onPress={this.goToChallengeDetailScene}>
+            <View style={styles.icon}>
+              <Text style={styles.goFab}>GO</Text>
+            </View>
+          </TouchableHighlight>
+        </View>
+        <Text style={[styles.caption,styles.description]}>{this.props.challenge.description}</Text>
+        <Text style={[styles.caption,styles.description]}>{this.state.completedBy}</Text>
+        <Text style={[styles.caption,styles.createdBy]}>{this.state.createdBy}</Text>
```

Comment:
Agreed. I merged it anyway since it appears to be working, but this is not how state is used. If we get a little bit of free time, let's refactor that. Essentially the text never changes so you replace {this.state.completedBy}, with {"Created by " + props.challenge.owner.displayName + " on " + props.challenge.created}. Storing things on the state object has added overhead because it is set up to use listeners so that it can re-render when state changes. This data is read-only and will never change so state is unnecessary.